### PR TITLE
PHP 8: clean up some of `Toolkit` package

### PR DESCRIPTION
**Pull request by distantnative**
**⚠️ Requires PHP 8, don't merge before Kirby 3.8**

### CODE REVIEW

File: src/Toolkit/Component.php [hunk 1]
```diff
@@ -21,69 +21,49 @@ class Component
...
+	protected array $methods = [];
 
 	/**
 	 * An array of all component options
 	 * from the component definition
-	 *
-	 * @var array
 	 */
-	protected $options = [];
+	protected array|string $options = [];
```
Comment (afbora) [inline src/Toolkit/Component.php:52]: I couldn't find any `string` usage?

File: src/Toolkit/Date.php [hunk 2]
```diff
@@ -264,9 +248,8 @@ public function month(int|null $month = null): int
 	 * Returns the datetime which is nearest to the object's datetime
 	 *
 	 * @param string|int|\DateTimeInterface ...$datetime Datetime strings, UNIX timestamps or objects
-	 * @return string|int|\DateTimeInterface
 	 */
-	public function nearest(...$datetime)
+	public function nearest(...$datetime): string|int|DateTimeInterface
```
Comment (afbora) [inline src/Toolkit/Date.php]: Is there any harm in using it like this?

```suggestion
	public function nearest(string|int|DateTimeInterface ...$datetime): string|int|DateTimeInterface
```

File: src/Toolkit/I18n.php [hunk 3]
```diff
@@ -130,14 +110,12 @@ public static function locale(): string
...
-	 * @param string|null $locale
-	 * @return string|array|null
 	 */
-	public static function translate($key, $fallback = null, string $locale = null)
-	{
+	public static function translate(
+		string|array|null $key,
+		string|array|null $fallback = null,
+		string|null $locale = null
+	): mixed {
```
Comment (afbora) [inline src/Toolkit/I18n.php:114]: Is it possible to return a value other than `string|array|null` types?

File: src/Toolkit/Str.php [hunk 4]
```diff
@@ -1132,9 +1056,6 @@ public static function startsWith(string $string = null, string $needle, bool $c
 	/**
 	 * Converts a string to studly caps case
 	 * @since 3.7.0
-	 *
-	 * @param string $value The string to convert
-	 * @return string
 	 */
 	public static function studly(string $value = null): string
```
Comment (afbora) [inline src/Toolkit/Str.php]:
```suggestion
	public static function studly(string|null $value = null): string
```

File: src/Toolkit/I18n.php (hunk 3 above)
Comment (distantnative) [reply src/Toolkit/I18n.php:114]: Yes, e.g. in `I18nTest.php:360` it would be a Closure.

File: src/Toolkit/Component.php (hunk 1 above)
Comment (distantnative) [reply src/Toolkit/Component.php:52]: Fields\TagsFieldTest::testOptionsQuery

Comment (afbora) [reply src/Toolkit/Component.php:52]: Are you sure and could you mixing `Component::$options` with `\Kirby\Form\Options`? Or am I mixing it up. If you're sure, it's fine.

I've run `testOptionsQuery` unit test and `Component::$options` still array.

Comment (distantnative) [reply src/Toolkit/Component.php:52]: You can see my commute. Had changed it but then our unit tests failed. Then reverted it.

Comment (afbora) [reply src/Toolkit/Component.php:52]: 🙈 Sorry, missed that!

**Review by afbora — DISMISSED** [review]
LGTM 👍

**Comment by distantnative** [comment]
@afbora added a few more fixes https://github.com/getkirby/kirby/pull/4527/commits/85d61c73f05d91bab914c8551004e7ba7eae346c

**Comment by afbora** [comment]
Looks good me but still it would be nice to have Lukas take a look at the psalm related changes.

**Comment by distantnative** [comment]
Will reopen once merge conflicts etc are solved

**Pull request closed without merging**